pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -209,13 +222,38 @@ func extractLibInfo(pod *corev1.Pod, containerRegistry string) []libInfo {
 			for _, lang := range supportedLanguages {
 				libInfoList = append(libInfoList, libInfo{
 					lang:  lang,
-					image: fmt.Sprintf(imageFormat, containerRegistry, lang, version),
+					image: libImageName(containerRegistry, lang, version),
 				})
 			}
 		}
 	}
 
-	return libInfoList
+	return libInfoList, autoDetected
+}
+
+// extractLibrariesFromOwnerAnnotations constructs the libraries to be injected if the languages
+// were stored in workloadmeta store based on owner annotations (for example: Deployment, Daemonset, Statefulset).
+func extractLibrariesFromOwnerAnnotations(pod *corev1.Pod, registry string) []libInfo {
+	libList := []libInfo{}
+
+	ownerName, ownerKind, found := getOwnerNameAndKind(pod)
+	if !found {
+		return libList
+	}
+
+	// TODO [Workloadmeta][Component Framework]: Use workloadmeta store as a component
+	store := workloadmeta.GetGlobalStore()
+	if store == nil {
+		return libList
+	}
+
+	// Currently we only support deployments
+	switch ownerKind {
+	case "Deployment":
+		libList = getLibListFromDeploymentAnnotations(store, ownerName, pod.Namespace, registry)
+	}
```

Comment:
maybe we can add a debug log in case we received the information on a ownerKind now yes supported.

```suggestion
	default:
		log.Debugf("This ownerKind:%s is not yet supported by the process language auto-detection feature", ownerKind)
	}
```

Reply:
I think we might as well write to upgrade as normally annotations should be added automatically by the agent